File: XiaomiAI/src/main/java/org/example/xiaomiai/entity/Conversation.java
package org.example.xiaomiai.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Conversation {
    @TableId(type = IdType.AUTO)
    private Integer id;
    private Integer sessionId;
    private Integer userId;
    private Integer role;  // 0为ai，1为用户
    private String content;
    private Long time;
}

File: xiaomi-ai-frontend/src/api/conversation.js
import axios from "./request";

export function selectConversation(sessionId){
  return axios.get("/conversation?sessionId=" + sessionId);
}

export function selectConversationByUserId(conversation,start,end){
  var queryUrl = "/conversation/"+start+"/"+end+"?userId=" + conversation.userId;
  if(conversation.sessionId != null){
    queryUrl += "&sessionId=" + conversation.sessionId;
  }
  if(conversation.role  != null){
    queryUrl += "&role=" + conversation.role;
  }
  if(conversation.content != null){
    queryUrl += "&content=" + conversation.content;
  }
  return axios.get(queryUrl);

}